### DISCUSSION TITLE
"ClientError: signOut called before signIn call" and "Cannot read properties of undefined (reading 'match')" error

### DISCUSSION THREAD

**Discussion by hungpham1406** [GogoVega/node-red-contrib-firebase-realtime-database#77] [Q&A]
Hello, after I have config the "Firebase In" node using JSON private key as instructed, but I still got the error "ClientError: signOut called before signIn call" and "Cannot read properties of undefined (reading 'match')". Does anyone know how to solve this?

**Comment by GogoVega** [comment]
Hi,
Thank you for your message. I will investigate 🙂
Could you please send me the full error message and a procedure to reproduce your error (what did you do to get the error).

**Reply by hungpham1406** [reply]
I use the node "Firebase In" to fetch data from my realtime db, then I drag my JSON private key into the node, I see the node has recognized my project Id, then I input the URL of realtime db. After finishing set up, I deployed and the node got error. You can view my set up and error in the pictures.
<img width="254" alt="Screenshot 2024-11-20 143233" src="https://github.com/user-attachments/assets/0bc7939e-f463-4b0f-a967-594e1ec4618e">
<img width="323" alt="Screenshot 2024-11-20 143204" src="https://github.com/user-attachments/assets/3c8b9c7f-2976-4396-833b-932a40721c27">
<img width="329" alt="Screenshot 2024-11-20 143133" src="https://github.com/user-attachments/assets/3222c23c-2357-4541-b0cf-810b2a855851">

<img width="278" alt="Screenshot 2024-11-20 143324" src="https://github.com/user-attachments/assets/09a65f66-81db-49ff-b42c-1cf04c4e6c5f">

**Reply by GogoVega** [reply]
I would need the complete error from the runtime (backend) log. The one that starts with welcome to Node-RED.

**Reply by hungpham1406** [reply]
You mean this one right?
![image](https://github.com/user-attachments/assets/c83d63f2-96f6-4b36-b709-77d27507b982)

**Reply by GogoVega** [reply]
Yes, unfortunately I don't have the trace but it doesn't matter I see better 😉 

Can you fill the `Client Email` and `Private Key` fields manually then deploy to see if that is the problem.

**Reply by GogoVega** [reply]
Errors are thrown because `credentials.privateKey` is undefined. The question is why?

The editor does not overwrite the value when saving the config node. So for me it's the content of the JSON file. Can you confirm that your file contains a property `privateKey` or `private_key`.
